fix: hide content with invalid date config in live mode

hassanazam opened the pull request:
## Summary
  - Fixed Page API returning content with invalid date configuration in live mode
  - Added validation to filter content where expireDate is before publishDate
  - Content remains visible in preview/edit modes for administrators to fix

  ## Changes Made
  - Modified `PageRenderUtil.populateContainers()` to skip invalid content in live mode
  - Added `hasInvalidDateConfiguration()` method for date validation
  - Created comprehensive unit tests for date validation logic
  - Added integration test for Page API behavior

  ## Test Plan
  - [x] Unit tests pass for all date validation scenarios
  - [x] Integration tests verify Page API filters invalid content in live mode
  - [x] Content with invalid dates still visible in preview mode
  - [x] Build and pre-commit hooks pass

  ## Related Issue
  Fixes #32261

  The branch contains:
  - 1 commit: fix: hide content with invalid date config in live mode
  - 4 files changed: PageRenderUtil.java (main fix) + 3 test files
  - All tests passing and pre-commit hooks successful

hassanazam committed:
fix: hide content with invalid date config in live mode

Content with expireDate before publishDate should not show in live mode.
Added validation in PageRenderUtil to filter such content.
Still visible in preview/edit mode for admins to fix.

Fixes #32261

hassanazam committed:
refactor: improve test quality for PageRenderUtil date validation

- Remove duplicate PageRenderUtilBugReproductionTest class with issue-specific naming
- Enhance PageRenderUtilDateValidationTest with better documentation and GitHub issue context
- Add PageMode integration test to validate LIVE vs PREVIEW mode behavior
- Consolidate test coverage in properly-named test class following codebase conventions

fabrizzio-dotCMS committed:
Merge branch 'main' into bugfix/32261-page-api-returns-expired-content

fabrizzio-dotCMS commented on dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java:
```diff
@@ -300,6 +300,13 @@ private List<ContainerRaw> populateContainers() throws DotDataException, DotSecu
 
+                    // Skip content with invalid date configuration in LIVE mode
+                    if (mode.showLive && hasInvalidDateConfiguration(contentlet)) {
+                        Logger.debug(this, ()-> "Skipping contentlet " + contentlet.getIdentifier()
```
I think it makes more sense to make this a warning, saying that we have ignored or skipped a piece of content that has an invalid config

fabrizzio-dotCMS replied:
perhaps the warning must be displayed for every piece of content that has an invalid configuration regardless the mode And skip it only for live mode

fabrizzio-dotCMS replied:
also I thik this exclusion logic is the first thing that needs to happen here 
e.g before this.addContentletPageReferenceCount(contentlet);

fabrizzio-dotCMS commented on dotcms-integration/src/test/java/com/dotcms/rest/api/v1/page/PageResourceTest.java:
```diff
@@ -1958,4 +1962,180 @@ public void Test_Rendering_Working_Content_Using_Limited_User() throws Exception
+    /**
+     * Test for GitHub Issue #32261: Content with invalid date configuration should be filtered in LIVE mode.
+     */
+    @Test
```
We try to doc our test in such a way that we describe the scenario and the expected results as follows:
Give that a contentlet with invalid dates rage set up yadda yadda yadda 
Expected: The Page Resource should render contents with a valid date range and skip invalid yadda yadda yadaa

jdotcms reviewed (approved):
ok in general, not fan of the continue but it is ok

hassanazam committed:
refactor: optimize date validation in PageRenderUtil

- Move date validation check before addContentletPageReferenceCount for efficiency
- Change log level from debug to warn for better production visibility
- Make hasInvalidDateConfiguration package-private for testability
- Add clearer warning message indicating specific date issue

Addresses PR review feedback from @fabrizzio-dotCMS

hassanazam committed:
test: remove reflection and improve PageRenderUtilDateValidationTest

- Remove reflection-based testing approach for cleaner test code
- Add static helper method with detailed explanation of approach
- Simplify test setup by removing unnecessary reflection boilerplate
- Maintain same test coverage with direct logic validation

Addresses PR review feedback from @fabrizzio-dotCMS about removing reflection usage

hassanazam committed:
docs: improve test documentation in PageResourceTest

- Add structured JavaDoc with HTML formatting following codebase patterns
- Use Given/Expected format as requested in PR review
- Include method reference and clear scenario description
- Document GitHub issue #32261 context for future reference

Addresses PR review feedback from @fabrizzio-dotCMS about test documentation format

fabrizzio-dotCMS commented on dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java:
```diff
@@ -782,4 +789,31 @@ public Set<? extends Map.Entry<String, List<String>>> entrySet() {
+    private boolean hasInvalidDateConfiguration(final Contentlet contentlet) {
+        final ContentType contentType = contentlet.getContentType();
+        
+        // Only check content types that have both date fields configured
```
I agree it is more readable, and debugger friedly 
 I think it's fine!

nollymar commented:
This issue has been fixed on [this PR](https://github.com/dotCMS/core/pull/32406)
